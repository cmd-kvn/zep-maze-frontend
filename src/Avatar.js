import React from 'react';
import {emojify} from 'react-emojione';
import { UP, DOWN, LEFT, RIGHT } from './Controller.constants';

export function Avatar({ stance }) {
    switch(stance) {
        case UP: return ( <div> emojify( ':point_up_tone3:' ) </div> );
        case DOWN: return ( <div> emojify( ':point_down_tone3:' ) </div> );
        case RIGHT: return ( <div> emojify(':point_right_tone3:') </div> );
        case LEFT: return ( <div> emojify( ':point_left_tone3:' ) </div> );
        default: return ( <div> emojify( ':levitate_tone3:' ) </div>);
    }
}