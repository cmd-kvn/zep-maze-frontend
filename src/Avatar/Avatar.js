import React from 'react';
import  Emojify from 'react-emojione';
import { connect } from 'react-redux';
import styles from './Avatar.css'

export default function Avatar() {
    
    return ( <div className={styles.avatar}> <Emojify><span>:wink:</span></Emojify> </div>);
}
