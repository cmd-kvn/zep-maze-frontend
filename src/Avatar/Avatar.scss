

:local(.avatar){
    // margin-left: -16px;
    // margin-top: -16px;
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    padding: auto;
    display: flex;   
    justify-content: center;
    align-items: center;
}