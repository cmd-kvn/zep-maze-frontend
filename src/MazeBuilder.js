import React from 'react';
import { Tile } from './Tile';

export function MazeBuilder({ matrix }) {
    return (
        <table>
            {matrix.map( ( row, index ) => {
                return (
                    <tr key={index}>
                        {row.map( ( tile, index ) => {
                            return <Tile key={index} value={tile}/>
                        })}
                    </tr>
                )
            })}
        </table>
    )
}