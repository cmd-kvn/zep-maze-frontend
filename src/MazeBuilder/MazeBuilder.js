import React, { Component } from 'react';
import { Tile } from '../Tile/Tile';
import { connect } from 'react-redux';

export class MazeBuilder extends Component{

    constructor(props){
        super(props);
    }    
    
    // componentDidMount() {
    //     fetch('/api/mazes')
    //         .then(res => res.json())
    //         .then(mazes => mazes[0].matrix = this.state.matrix);
    // }

    render() {

        const { x, y, matrix } = this.props;

        const xVal = x * 40;
        const yVal = y * 40
        
        const styles = {
            transform: `transform: translate( ${ xVal }px, ${ yVal }px )`
        }
        return (
            <table className={"Maze"} style={styles}>
                {matrix.map( ( row, index ) => {
                    return (
                        <tr key={index}>
                        {row.map( ( tile, index ) => {
                            return <Tile key={index} value={tile}/>
                        })}
                        </tr>
                    )
                })}
            </table>
    )}
}

export default connect(state => ({
    matrix: state.matrix,
    x: state.x,
    y: state.y
}))(MazeBuilder);