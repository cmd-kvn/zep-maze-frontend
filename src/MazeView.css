:local(.viewer) {
  overflow: hidden;
  display: flex; }
