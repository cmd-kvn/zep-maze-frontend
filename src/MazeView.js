import React from 'react';
import styles from './MazeView.css';


export function MazeView(props) {
    return (
        <div className={styles.viewer}>
            {props.children}
        </div>
    )
}