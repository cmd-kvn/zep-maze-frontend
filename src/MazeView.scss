:local(){
    .local{
        width: 120;
        height: 120;
        overflow: hidden;
        display: flex;   
    }

    .Avatar{
        justify-content: center;
    }
}