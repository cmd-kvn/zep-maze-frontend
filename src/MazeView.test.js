import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MazeView } from './MazeView';
import { Avatar } from './Avatar';
import { MazeBuilder } from './MazeBuilder';

describe('MazeView component', () => {
    it('renders a div when no child components passed', () => {
        const wrapper = shallow(<MazeView/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})