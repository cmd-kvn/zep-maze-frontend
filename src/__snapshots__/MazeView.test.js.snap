// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MazeView component renders a div when no child components passed 1`] = `<div />`;
